pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -2110,3 +2057,47 @@ func GetVolumeNameByTarget(domain *api.Domain, target string) string {
 	}
 	return ""
 }
+
+func DetectEFIConfig(arch, ovmfPath string, secureBoot bool) (*EFIConfiguration, error) {
+	secureLoader := "yes"
+	efiCode := EFICode
+	efiVars := EFIVars
+
+	if !secureBoot {
+		secureLoader = "no"
+	}
+
+	if secureBoot {
+		efiCode = EFICodeSecureBoot
+		efiVars = EFIVarsSecureBoot
+	} else if isARM64(arch) {
+		efiCode = EFICodeAARCH64
+		efiVars = EFIVarsAARCH64
+	}
+
+	_, err := os.Stat(filepath.Join(ovmfPath, efiCode))
+	if os.IsNotExist(err) && efiCode == EFICode {
+		// The combination (EFICodeSecureBoot + EFIVars) is valid
+		// for booting in EFI mode with SecureBoot disabled
+		efiCode = EFICodeSecureBoot
+		_, err = os.Stat(filepath.Join(ovmfPath, efiCode))
```

Comment:
Would it make sense to handle this `err` inside the `if`? It is easy to mess up things which error-reusing.

Reply:
I see your point, I've done it this way just to not repeat the same error handling twice and I have no strong preference, if you think it improves maintainability I'll change it, what do you think?

Reply:
I am fine.